model/example_domain_test.go

```diff
@@ -0,0 +1,618 @@
+package model_test
+
+import (
+	"fmt"
+
+	"github.com/nextmv-io/sdk/model"
+)
+
+func ExampleNewDomain_empty() {
+	domain := model.NewDomain()
+
```

Comment:
I will only make this comment once, but it applies to all tests. All the tests that are already in place do not introduce newline before the `fmt.Println` statements. Is it ok of we keep consistency here so all the codebase of examples feels similar? I'm appending an [example from the `store` package](https://github.com/nextmv-io/sdk/blob/a0dbeac8f5d41e65e8fdb59cbcd679d533b77b41/store/example_domain_test.go#L10-L41) so that you can see what I mean:

```go
func ExampleDomain_add() {
	s1 := store.New()
	d := store.Multiple(s1, 1, 3, 5)
	s2 := s1.Apply(d.Add(2, 4))
	fmt.Println(d.Domain(s1))
	fmt.Println(d.Domain(s2))
	// Output:
	// {[{1 1} {3 3} {5 5}]}
	// {[{1 5}]}
}


func ExampleDomain_atLeast() {
	s1 := store.New()
	d := store.NewDomain(s1, model.NewRange(1, 10), model.NewRange(101, 110))
	s2 := s1.Apply(d.AtLeast(50))
	fmt.Println(d.Domain(s1))
	fmt.Println(d.Domain(s2))
	// Output:
	// {[{1 10} {101 110}]}
	// {[{101 110}]}
}


func ExampleDomain_atMost() {
	s1 := store.New()
	d := store.NewDomain(s1, model.NewRange(1, 10), model.NewRange(101, 110))
	s2 := s1.Apply(d.AtMost(50))
	fmt.Println(d.Domain(s1))
	fmt.Println(d.Domain(s2))
	// Output:
	// {[{1 10} {101 110}]}
	// {[{1 10}]}
}
```